File: BAEKJOON/Javascript/10000/10800/10811.js
/*
Title : 바구니 뒤집기
Link : https://www.acmicpc.net/problem/10811
*/

const fs = require("fs");
const filepath =
  process.platform === "linux" ? "/dev/stdin" : __dirname + "/input.txt";
const input = fs.readFileSync(filepath).toString().trim().split("\n");

const [n, m] = input[0].split(" ").map(Number);
const basket = Array.from({ length: n }, (_, i) => i + 1);

for (let i = 0; i < m; i++) {
  const [start, end] = input[i + 1].split(" ").map(Number);
  const temp = basket.slice(start - 1, end).reverse();
  basket.splice(start - 1, end - start + 1, ...temp);
}

console.log(basket.join(" "));

File: BAEKJOON/Python/10000/10800/10811.py
"""
title : 바구니 뒤집기
link : https://www.acmicpc.net/problem/10811
"""

from sys import stdin

input = stdin.readline


if __name__ == "__main__":
    n, m = map(int, input().split())
    basket = [i for i in range(1, n + 1)]
    for _ in range(m):
        i, j = map(int, input().split())
        basket[i - 1 : j] = basket[i - 1 : j][::-1]
    print(*basket)
